[macOS] Add tcl-tk to macOS13

sarathrajsrinivasan opened the pull request:
# Description
Setup-actions team currently use Tcl/Tk in macOS12 to generate artifacts for python. Since macOS12 is deprecated adding Tcl/Tk to macOS13.

#### Related issue:

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

erik-bershel commented:
- Lines 289 and 297 has duplicated functionality. 
- Miscellaneous section seems a bit messy in general.
- If this is an immutable dependency of the underlying service, why not extend the practice to the entire spectrum of Intel-based macOS?

sarathrajsrinivasan commented:
> * Lines 289 and 297 has duplicated functionality.
> * Miscellaneous section seems a bit messy in general.
> * If this is an immutable dependency of the underlying service, why not extend the practice to the entire spectrum of Intel-based macOS?

- Lines 288 to 294 are specific to Monterey and can be removed when macOS12 is deprecated.
- Rewrote the Miscellaneous section.
- Added support for both macOS13 and macOS14.
- Removed `IsSequoiaX64` as there are no specific requirements for macOS15 yet under Miscellaneous.
